src/Components/Components/src/NavigationManagerExtensions.cs

```diff
@@ -0,0 +1,694 @@
+            => value?.ToString(CultureInfo.InvariantCulture);
+
+        private static string Format(double value)
+            => value.ToString(CultureInfo.InvariantCulture);
+
+        private static string? Format(double? value)
+            => value?.ToString(CultureInfo.InvariantCulture);
+
+        private static string Format(float value)
+            => value.ToString(CultureInfo.InvariantCulture);
+
+        private static string? Format(float? value)
+            => value?.ToString(CultureInfo.InvariantCulture);
+
+        private static string Format(Guid value)
+            => value.ToString(null, CultureInfo.InvariantCulture);
+
+        private static string? Format(Guid? value)
+            => value?.ToString(null, CultureInfo.InvariantCulture);
+
+        private static string Format(int value)
+            => value.ToString(CultureInfo.InvariantCulture);
+
+        private static string? Format(int? value)
+            => value?.ToString(CultureInfo.InvariantCulture);
+
+        private static string Format(long value)
+            => value.ToString(CultureInfo.InvariantCulture);
+
+        private static string? Format(long? value)
+            => value?.ToString(CultureInfo.InvariantCulture);
+
+        private struct QueryStringBuilder
+        {
+            private readonly StringBuilder _builder;
+
+            private bool _hasNewParameters;
+
+            public string UriWithQueryString => _builder.ToString();
+
+            public QueryStringBuilder(ReadOnlySpan<char> uriWithoutQueryString)
+            {
+                _builder = new();
+                _builder.Append(uriWithoutQueryString);
+
+                _hasNewParameters = false;
+            }
+
+            public void AppendParameter(ReadOnlySpan<char> encodedName, ReadOnlySpan<char> encodedValue)
+            {
+                if (!_hasNewParameters)
+                {
+                    _hasNewParameters = true;
+                    _builder.Append('?');
+                }
+                else
+                {
+                    _builder.Append('&');
+                }
+
+                _builder.Append(encodedName);
+                _builder.Append('=');
+                _builder.Append(encodedValue);
+            }
+        }
+
+        private class ParameterData
+        {
+            public string? EncodedValue { get; private set; }
+            public bool DidReplace { get; set; }
+
+            public ParameterData(string? encodedValue)
+            {
+                EncodedValue = encodedValue;
+                DidReplace = false;
+            }
+        }
+
+        /// <summary>
+        /// Returns a <see cref="string"/> equal to <see cref="NavigationManager.Uri"/> except with a single parameter
+        /// added or updated.
+        /// </summary>
+        /// <param name="navigationManager">The <see cref="NavigationManager"/>.</param>
+        /// <param name="name">The name of the paramter to add or update.</param>
```

Comment:
```suggestion
        /// <param name="name">The name of the parameter to add or update.</param>
```

Same typo in other places below.

Reply:
Ah, thanks! I kept making this typo but thought I fixed them all - I guess I missed some 😄